tests/pootle_fs/resources.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+import pytest
+
+from pootle_fs.models import StoreFS
+from pootle_fs.resources import FSResources
+from pootle_project.models import Project
+from pootle_store.models import Store
+
+
+@pytest.mark.django_db
+def test_project_resources_instance():
+    project = Project.objects.get(code="project0")
+    resources = FSResources(project)
+    assert resources.project == project
+    assert str(resources) == "<FSResources(Project 0)>"
+
+
+@pytest.mark.django_db
+def test_project_resources_stores():
+    project = Project.objects.get(code="project0")
+    stores = Store.objects.filter(
+        translation_project__project=project)
+    assert list(FSResources(project).stores) == list(stores)
+    # mark some Stores obsolete - should still show
+
+
+@pytest.mark.django_db
+def test_project_resources_trackable_stores(project0_fs_resources):
+    project = project0_fs_resources
+    stores = Store.objects.filter(
+        translation_project__project=project)
+    # only stores that are not obsolete and do not have an
+    # exiting StoreFS should be trackable
+    trackable = stores.filter(obsolete=False).order_by("pk")
+    trackable = trackable.filter(fs__isnull=True)
+    assert (
+        list(FSResources(project).trackable.order_by("pk"))
+        == list(trackable))
+    for store in FSResources(project).trackable:
+        try:
+            fs = store.fs.get()
+        except StoreFS.DoesNotExist:
+            fs = None
+        assert fs is None
+        assert store.obsolete is False
+
+
+@pytest.mark.django_db
+def test_project_resources_tracked(project0_fs_resources):
+    project = project0_fs_resources
+    assert (
+        list(FSResources(project).tracked.order_by("pk"))
+        == list(StoreFS.objects.filter(project=project).order_by("pk")))
+    # this includes obsolete stores
+    assert FSResources(project).tracked.filter(
+        store__obsolete=True).exists()
+
+
+@pytest.mark.django_db
+def test_project_resources_synced(project0_fs_resources):
+    project = project0_fs_resources
+    synced = StoreFS.objects.filter(project=project).order_by("pk")
+    obsoleted = synced.filter(store__obsolete=True).first()
+    obsoleted.last_sync_hash = "FOO"
+    obsoleted.last_sync_revision = 23
+    obsoleted.save()
+    active = synced.exclude(store__obsolete=True).first()
+    active.last_sync_hash = "FOO"
+    active.last_sync_revision = 23
+    active.save()
+    synced = synced.exclude(last_sync_revision__isnull=True)
+    synced = synced.exclude(last_sync_hash__isnull=True)
```

Comment:
i did it this way for a reason, that being that the ORM can do different things with in these 2 situations - altho not always clear to me exactly why.

basically i want an OR exclude, and im not 100% i would get that with the way you show